File: Programming_lab-06/Programming_lab-06/SupplierOfBook.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Programming_lab_06
{
    class SupplierOfBook: SupplierOfGoods
    {
        public override bool acceptOrder(Shop shop, string name, int quantity)
        {
            int n = Contracts.Count;
            int i = 0;
            for (; i < n; i++)
                if (Contracts[i].shop_ == shop) n = -1;

            bool ret = false;
            if (n < 0)
            {
                ret = true;
                Contracts[i - 1].orders_.Add(new Order(name, quantity));
            }
            return ret;
        }
    }
}

File: Programming_lab-05/src/SupplierOfBook.java

public class SupplierOfBook extends SupplierOfGoods
{
	@Override
	public boolean acceptOrder(Shop shop, String name, int quantity)
	{      
		boolean ret = false;
		
		if(name.indexOf("Book:") == 0)
		{
	        int n = getContracts().size();
	        int i = 0;
	        for(; i < n; i++)            
	            if (getContracts().get(i).shop_ == shop) n = -1;
	
	        
	        if (n < 0)
	        {
	        	getContracts().get(i - 1).orders_.add(new Order(name, quantity));
	        }
		}
        return ret;            
    }

}
